### PR TITLE
Apply NewLine in StreamWriter, improve performance

### PR DESCRIPTION
**Pull request by patagonaa**
## Description

- use the NewLine that may be set on StreamWriter in WriteLine()/WriteLine(string).
- possibly improve WriteLine()/WriteLine(string) performance (skip String append + Encoding.GetBytes() for newline characters).
- improve Write(string) performance (write string in one go instead of char by char).

## Motivation and Context

- Fixes nanoFramework/Home#1458

## How Has This Been Tested?

Created a new NF App, referenced the local version of the library, ran example code, looked at result in debugger
```csharp
using var ms = new MemoryStream();
var sw = new StreamWriter(ms) { NewLine = "\n" };
sw.WriteLine("Test");
sw.Write("Test");
sw.Flush();

var outBytes = ms.ToArray();
var output = Encoding.UTF8.GetString(outBytes, 0, outBytes.Length);
// output is "Test\nTest"
```

Would've loved to add some Unit Tests as well, but seeing there aren't any to begin with (and I don't really know how they're supposed to look like), I didn't add any.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by Ellerbach** [comment]
Thanks @patagonaa the PR is failing but it's not on you. Looking at this.

**Review by Ellerbach — APPROVED** [review]
Thanks for the improvement and bug fix. It looks good. Let's wait for the pipeline to be fixed (it's on us).

File: System.IO.Streams/StreamWriter.cs [hunk 1]
```diff
@@ -13,12 +13,12 @@ namespace System.IO
     /// </summary>
     public class StreamWriter : TextWriter
     {
-        private const string c_NewLine = "\r\n";
         private const int c_BufferSize = 0xFFF;
 
         private bool _disposed;
         private byte[] _buffer;
         private int _curBufPos;
+        private byte[] _newLineBytes = Encoding.UTF8.GetBytes("\r\n"); // must equal TextWriter._InitialNewLine
```
Comment (josesimoes) [inline System.IO.Streams/StreamWriter.cs:21]: Per project code style: comments above the code please.

**Review by josesimoes — CHANGES_REQUESTED** [review]

**Comment by josesimoes** [comment]
@patagonaa regarding unit tests: that would be awesome!
Please follow the pattern that is used in any of the other class libs that has them. For example: https://github.com/nanoframework/System.IO.Ports/tree/main/Tests/UnitTestsSerialPort

**Review by josesimoes — APPROVED** [review]
Perfect! Thanks

**Pull request merged**